cloud/amazon/iam_saml_federation.py

```diff
@@ -0,0 +1,185 @@
+  state:
+    description:
+      - Whether to create or delete identity provider.  If 'present' is specified it will attempt to update the identity provider matching the name field.
+    default: present
+    choices: [ "present", "absent" ]
+extends_documentation_fragment:
+ - aws
+author: "Aidan Rowe (@aidan-)"
+'''
+
+EXAMPLES = '''
+# Note: None of these examples set aws_access_key or aws_secret_key.
+# It is assumed that their matching environment variables are set.
+
+# Creates a new iam saml identity provider if not present
+- name: saml provider
+  iam_saml_federation:
+    name: example1
+    saml_metadata_document: >
+  <?xml version="1.0"?>...
+  <md:EntityDescriptor  
+
+# Creates a new iam saml identity provider if not present
+- name: saml provider
+  iam_saml_federation:
+    name: example2
+    saml_metadata_document: "{{ item }}"
+  with_file: /path/to/idp/metdata.xml
+
+# Removes iam saml identity provider
+- name: remove saml provider
+  iam_saml_federation:
+    name: example3
+    state: absent
+'''
+
+RETURN = '''
+'''
+
+try:
+  import boto3
+  import botocore.exceptions
+  HAS_BOTO3 = True
+except ImportError:
+  HAS_BOTO3 = False
+import sys
+
+class SAMLProviderManager:
+  """Handles SAML Identity Provider configuration"""
+
+  def __init__(self, module, **aws_connection_params):
+    self.module = module
+    self.aws_connection_params = aws_connection_params
+
+    try:
+      self.conn= boto3_conn(module, conn_type='client', resource='iam', **self.aws_connection_params)
+    except botocore.exceptions.ClientError as e:
+      self.module.fail_json(msg=str(e))
+
+  def _get_provider_arn(self, name):
+    provider_arn = ""
+    providers = self.conn.list_saml_providers()
+    for p in providers['SAMLProviderList']:
+      provider_name = p['Arn'].split('/', 1)[1]
+      if name == provider_name:
+        return p['Arn']
+
+    return None
+
+  def create_or_update_saml_provider(self, name, metadata):
+    if not metadata:
+      self.module.fail_json(changed=changed, msg="saml_metadata_document must be defined for present state")
+
+    res = { 'changed' : False }
+    arn = self._get_provider_arn(name)
+    if arn: #see if metadata needs updating
+      resp = self.conn.get_saml_provider(SAMLProviderArn=arn)
+
+      if metadata.strip() != resp['SAMLMetadataDocument'].strip():
+        # provider needs updating
+        res['changed'] = True
+        if not self.module.check_mode:
+          try:
+            resp = self.conn.update_saml_provider(SAMLProviderArn=arn, SAMLMetadataDocument=metadata)
+          except botocore.exceptions.ClientError as e:
+            res['msg'] = str(e)
+            res['debug'] = [arn, metadata]
+            self.module.fail_json(**res)
+
+    else: #create
+      res['changed'] = True
+      if not self.module.check_mode:
+        try:
+          resp = self.conn.create_saml_provider(SAMLMetadataDocument=metadata, Name=name)
```

Comment:
Can you return the provider ARN in both the update & create cases? `boto3` returns them as `SAMLProviderArn` on the response object. http://boto3.readthedocs.io/en/latest/reference/services/iam.html#IAM.Client.create_saml_provider